mindscope_utilities/general_utilities.py

```diff
@@ -104,14 +104,22 @@ def event_triggered_response(data, t, y, event_times, t_before=1, t_after=1, out
     # set up a dictionary with key 'time' and
     # value as a linearly spaced time array
     step_size = 1/output_sampling_rate
-    data_dict = {
-        'time': np.linspace(
-            -t_before,
-            t_after,
-            int((t_before + t_after) / step_size + int(include_endpoint)),
-            endpoint=include_endpoint
-        )
-    }
+    # define a time array
+    n_steps = (t_before + t_after) / step_size
+    if n_steps != int(n_steps):
+        # if the number of steps isn't an int, that means it isn't possible
+        # to end on the desired t_after using the defined sampling rate
+        # we need to round down and include the endpoint
+        n_steps = int(n_steps)
+        t_after = -1*t_before + n_steps*step_size
```

Comment:
I just noticed it. Should `t_before` be negative? I think it's potentially confusing to hard code it to be negative because it doesn't allow for options of the start time being after an event.

Reply:
This is a good point. I should have gotten more opinions on this before setting up this function. My thought was that, for example, `t_before = 0.5` would mean 500 ms before the event. `t_before = 0` would mean 0 seconds before the event. But there is really no good way to set t_before to be after the event. 

You'd have to pass `t_before = -0.5` to get the response to start 500 ms after the stimulus. I agree that's confusing. But I'm also concerned that changing it would break backward compatibility.

Reply:
I agree we don't want to have notebooks around that would not work with new code, but I also think that we will unlikely get everything right the first time, so changes are inevitable -  we just need to document them and keep track of them. It might be worth making those changes sooner rather than later.